Admin/ArticleAdmin.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ArticleBundle\Admin;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Show\ShowMapper;
+use Sonata\ArticleBundle\Model\Article;
+use Sonata\ArticleBundle\Model\Fragment;
+use Sonata\CoreBundle\Validator\ErrorElement;
+use Sonata\MediaBundle\Admin\ORM\MediaAdmin;
+
+/**
+ * @author Florent Denis <florent.denis@ekino.com>
+ */
+class ArticleAdmin extends AbstractAdmin
+{
+    /**
+     * @var array
+     */
+    protected $datagridValues = array('_sort_by' => 'updatedAt', '_sort_order' => 'DESC');
+
+    /**
+     * @var MediaAdmin
+     */
+    protected $mediaAdmin;
+
+    /**
+     * @var string
+     */
+    protected $fragmentClass;
+
+    /**
+     * @param string     $code
+     * @param string     $class
+     * @param string     $baseControllerName
+     * @param MediaAdmin $mediaAdmin
+     * @param            $fragmentClass
+     */
+    public function __construct($code, $class, $baseControllerName, MediaAdmin $mediaAdmin, $fragmentClass)
+    {
+        parent::__construct($code, $class, $baseControllerName);
+
+        $this->mediaAdmin = $mediaAdmin;
```

Comment:
This is used by the ``getMediaBuilder`` function, which is not used yet ;) We are migrating code from an actual code base.